java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaObjectInputStream.java

```diff
@@ -182,40 +184,68 @@ public int available() throws IOException {
         return backingStream.available();
     }
 
+    /**
+     * <p>Closes this stream.</p>
+     *
+     * <p>This is a special version of {@link #close close()} which prevents
+     * re-use of the underlying connection, if any. Calling this method
+     * indicates that there should be no attempt to read until the end of
+     * the stream.</p>
+     *
+     * <p>If the backing stream of the connection is not a
+     * {@link EofSensorInputStream}, this method will call {@link #close()}.
+     * This subsequently closes the input stream and releases any system resources associated
+     * with the stream.</p>
+     *
+     * @throws IOException thrown when unable to close connection
+     */
     @Override
     public void close() throws IOException {
         if (LOGGER.isTraceEnabled()) {
             LOGGER.trace("Closing backingStream {} and response {}",
                     this.backingStream, httpResponse);
         }
 
-        try {
+        if (backingStream instanceof EofSensorInputStream) {
+                ((EofSensorInputStream) backingStream).close();
+                closeHttpResponse();
+            } else {
+            try {
             if (backingStream != null) {
                 backingStream.close();
+                }
+            } catch (IOException e) {
+                /* Since we will be throwing an exception when closing the backing
+                * stream, we will only log the error thrown when closing the
+                * response because we can only throw up a single exception. */
+                closeHttpResponse();
```

Comment:
Isn't this call superfluous as `closeHttpResponse` will be called in the `finally` block regardless?

Reply:
Yes, I think I added it since `close()` can tolerate multiple calls, but maybe minimizing usage of calls will be the right thing to do so it will be wiser to just catch the exception and `LOG` a warning.